pkg/scalers/neo4j_scaler.go

```diff
@@ -0,0 +1,210 @@
+package scalers
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"strconv"
+
+	"github.com/go-logr/logr"
+	"github.com/neo4j/neo4j-go-driver/v5/neo4j"
+	v2 "k8s.io/api/autoscaling/v2"
+	"k8s.io/metrics/pkg/apis/external_metrics"
+
+	kedautil "github.com/kedacore/keda/v2/pkg/util"
+)
+
+type neo4jScaler struct {
+	metricType v2.MetricTargetType
+	metadata   *neo4jMetadata
+	driver     neo4j.DriverWithContext
+	logger     logr.Logger
+}
+
+type neo4jMetadata struct {
+	connectionString     string
+	host                 string
+	port                 string
+	username             string
+	password             string
+	query                string
+	queryValue           int64
+	activationQueryValue int64
+	metricName           string
+	scalerIndex          int
+}
+
+func (s *neo4jScaler) GetMetricSpecForScaling(context.Context) []v2.MetricSpec {
+	externalMetric := &v2.ExternalMetricSource{
+		Metric: v2.MetricIdentifier{
+			Name: GenerateMetricNameWithIndex(s.metadata.scalerIndex, s.metadata.metricName),
+		},
+		Target: GetMetricTarget(s.metricType, s.metadata.queryValue),
+	}
+	metricSpec := v2.MetricSpec{
+		External: externalMetric, Type: externalMetricType,
+	}
+	return []v2.MetricSpec{metricSpec}
+}
+
+func (s *neo4jScaler) Close(ctx context.Context) error {
+	if s.driver != nil {
+		err := s.driver.Close(ctx)
+		if err != nil {
+			s.logger.Error(err, fmt.Sprintf("failed to close neo4j connection, because of %v", err))
+			return err
+		}
+	}
+	return nil
+}
+
+func (s *neo4jScaler) getQueryResult(ctx context.Context) (int64, error) {
+	session := s.driver.NewSession(ctx, neo4j.SessionConfig{
+		AccessMode: neo4j.AccessModeWrite,
+	})
+	defer session.Close(ctx)
+	result, err := session.ExecuteWrite(ctx, matchItemFn(s.metadata.query, ctx))
```

Comment:
This is from an example from neo4j official golang client, they have suggested this.

Reply:
They have more use cases in their examples: https://github.com/neo4j-examples/movies-golang-bolt/blob/main/server.go

Reply:
They also use other methods for reading, I have found [this example](https://github.com/neo4j-examples/movies-golang-bolt/blob/main/server.go) and they read with `ReadTransaction`, We should test and avoid writing actions in KEDA when it's possible even they can also read.

Reply:
Sure, I can try that

Reply:
There is no `ReadTransaction` in v5, that example is v4

Reply:
but, is `session.ExecuteRead` available in v5? I haven't reviewed the whole SDK but I got it yesterday from one of their examples, I can find it again if you want, my concern is to allow writing operations in general,  doesn't matter if we use `session.ExecuteRead` or `session.ReadTransaction`  to avoid it

Reply:
Now, the query includes write functionality also, as it contains `create`, not only `match`

Reply:
So, would have to use `AccessModeWrite` and `ExecuteWrite`, both

Reply:
Yes, `session.ExecuteRead` is available in v5

Reply:
the question is: why do we need to create anything? I mean, end users should create whatever is needed and KEDA should just query

Reply:
Works with read functions